src/libraries/System.Data.Common/src/System/Data/Common/DataStorage.cs

```diff
@@ -123,8 +123,8 @@ internal abstract class DataStorage
         internal readonly bool _isStringType;
         internal readonly bool _isValueType;
 
-        private static readonly Func<Type, Tuple<bool, bool, bool, bool>> s_inspectTypeForInterfaces = InspectTypeForInterfaces;
-        private static readonly ConcurrentDictionary<Type, Tuple<bool, bool, bool, bool>> s_typeImplementsInterface = new ConcurrentDictionary<Type, Tuple<bool, bool, bool, bool>>();
+        private static readonly Func<Type, (bool, bool, bool, bool)> s_inspectTypeForInterfaces = InspectTypeForInterfaces;
```

Comment:
It could also be bitmasked byte

Reply:
It could.  But that's a larger change than I'd like to make here.  And if this code is being used, I don't expect the size impact is going to be a huge concern.  I'm also not confident in our testing of this area.

Reply:
(Turns out this file is also compiled into libraries that build for netstandard2.0, so I've reverted the changes here entirely.)